### PR TITLE
[CARBONDATA-784] Make configurable empty data to be treated as bad record or not And Expose BAD_RECORDS_ACTION default value to be configurable from out side.	

### PR DESCRIPTION
**Pull request by mohammadshahidkhan**
Make configurable empty data to be treated as bad record or not And Expose BAD_RECORDS_ACTION default value to be configurable from out side.
1. Currently carbon Empty data is considered as bad record.
A property in the load option should be provided so as
the user can tell weather to consider empty data as bad record or not.
IS_EMPTY_DATA_BAD_RECORD = false/true
*default to false *
For example if IS_EMPTY_DATA_BAD_RECORD false below data will not treated as bad record vice versa
"","",""
,,
*2.Expose BAD_RECORDS_ACTION default value to be configurable from out side. *
A property carbon.bad.records.action will be added in carbon properties 
so that user can configure default value of BAD_RECORDS_ACTION.
default to FAIL

### CODE REVIEW

**Comment by CarbonDataQA** [comment]
Build Failed  with Spark 1.6.2, Please check CI http://136.243.101.176:8080/job/ApacheCarbonPRBuilder/1169/

**Comment by CarbonDataQA** [comment]
Build Failed  with Spark 1.6.2, Please check CI http://136.243.101.176:8080/job/ApacheCarbonPRBuilder/1175/

**Comment by CarbonDataQA** [comment]
Build Failed  with Spark 1.6.2, Please check CI http://136.243.101.176:8080/job/ApacheCarbonPRBuilder/1177/

**Comment by CarbonDataQA** [comment]
Build Success with Spark 1.6.2, Please check CI http://136.243.101.176:8080/job/ApacheCarbonPRBuilder/1179/

**Comment by CarbonDataQA** [comment]
Build Success with Spark 1.6.2, Please check CI http://136.243.101.176:8080/job/ApacheCarbonPRBuilder/1180/

**Comment by CarbonDataQA** [comment]
Build Success with Spark 1.6.2, Please check CI http://136.243.101.176:8080/job/ApacheCarbonPRBuilder/1181/

**Comment by CarbonDataQA** [comment]
Build Success with Spark 1.6.2, Please check CI http://136.243.101.176:8080/job/ApacheCarbonPRBuilder/1184/

**Comment by CarbonDataQA** [comment]
Build Success with Spark 1.6.2, Please check CI http://136.243.101.176:8080/job/ApacheCarbonPRBuilder/1198/

File: processing/src/main/java/org/apache/carbondata/processing/newflow/converter/impl/RowConverterImpl.java [hunk 1]
```diff
@@ -164,6 +169,39 @@ public CarbonRow convert(CarbonRow row) throws CarbonDataLoadingException {
   }
 
   @Override
+  public RowConverter createCopyForNewThread() {
```
Comment (gvramana) [inline processing/src/main/java/org/apache/carbondata/processing/newflow/converter/impl/RowConverterImpl.java]: why this function is moved

Comment (mohammadshahidkhan) [inline processing/src/main/java/org/apache/carbondata/processing/newflow/converter/impl/RowConverterImpl.java]: Fixed

File: core/src/main/java/org/apache/carbondata/core/constants/CarbonCommonConstants.java [hunk 2]
```diff
@@ -185,8 +185,7 @@
   /**
    * CARBON_BADRECORDS_LOCATION_DEFAULT
    */
-  public static final String CARBON_BADRECORDS_LOC_DEFAULT_VAL =
-      "../unibi-solutions/system/carbon/badRecords";
+  public static final String CARBON_BADRECORDS_LOC_DEFAULT_VAL = "D:/tmp/carbon/badRecords";
```
Comment (mohammadshahidkhan) [inline core/src/main/java/org/apache/carbondata/core/constants/CarbonCommonConstants.java]: fixed made it to /tmp/carbon/badRecords

File: processing/src/main/java/org/apache/carbondata/processing/newflow/converter/impl/MeasureFieldConverterImpl.java [hunk 3]
```diff
@@ -57,11 +61,18 @@ public void convert(CarbonRow row, BadRecordLogHolder logHolder)
     String value = row.getString(index);
     Object output;
     boolean isNull = CarbonCommonConstants.MEMBER_DEFAULT_VAL.equals(value);
-    if (value == null || value.length() == 0 || isNull) {
+    if (value == null || isNull) {
       logHolder.setReason(
           "The value " + " \"" + value + "\"" + " with column name " + measure.getColName()
               + " and column data type " + dataType + " is not a valid " + dataType + " type.");
       row.update(null, index);
+    } else if (value.length() == 0) {
```
Comment (mohammadshahidkhan) [inline processing/src/main/java/org/apache/carbondata/processing/newflow/converter/impl/MeasureFieldConverterImpl.java]: both the condition could not be merged.
when value is empty there will be two decision to be take
1. empty is good 
2. empty data is bad

**Comment by CarbonDataQA** [comment]
Build Success with Spark 1.6.2, Please check CI http://136.243.101.176:8080/job/ApacheCarbonPRBuilder/1207/

**Comment by CarbonDataQA** [comment]
Build Success with Spark 1.6.2, Please check CI http://136.243.101.176:8080/job/ApacheCarbonPRBuilder/1209/

**Comment by gvramana** [comment]
LGTM

**Pull request closed without merging**